components/camel-azure/camel-azure-storage-blob/src/main/java/org/apache/camel/component/azure/storage/blob/BlobStreamAndLength.java

```diff
@@ -40,14 +40,16 @@ private BlobStreamAndLength(InputStream inputStream, long streamLength) {
     @SuppressWarnings("rawtypes")
     public static BlobStreamAndLength createBlobStreamAndLengthFromExchangeBody(final Exchange exchange) throws IOException {
         Object body = exchange.getIn().getBody();
+        Long blobSize = exchange.getIn().getHeader(BlobConstants.BLOB_UPLOAD_SIZE, () -> null, Long.class);
```

Comment:
You can maybe remove the header here instead?

Reply:
As the remove will give you back the header

Reply:
I was wondering about that as-well. It's at least better than the current spot, since it's in context. :+1:

Reply:
Change done.